phoenix-core/src/main/java/org/apache/phoenix/iterate/TableSnapshotResultIterator.java

```diff
@@ -164,9 +165,6 @@ public void close() throws SQLException {
     closed = true; // ok to say closed even if the below code throws an exception
     try {
       scanIterator.close();
-      fs.delete(this.restoreDir, true);
```

Comment:
@sakshamgangwar - It seems correct that if we're not restoring the snapshot in the iterator anymore that we can't clean it up there either. But it still seems like we need to clean up somewhere.

Reply:
@gjacoby126 You are right Geoffrey! For our use case, we do clean up at the post-processing step for this restore directory too. Let me try to find a better place here as well to do the cleanup.